From 22817982ca0061f1a2ae7e427ca7fce02bf47eae Mon Sep 17 00:00:00 2001
From: Don Dutile <ddutile@redhat.com>
Date: Thu, 9 Nov 2017 06:48:30 -0500
Subject: [PATCH] [netdrv] qed: fix missing break in OOO_LB_TC case

Message-id: <20171109065423.8574-116-ddutile@redhat.com>
Patchwork-id: 197094
O-Subject: [PATCH RHEL-7.5 v2 115/468] qed: fix missing break in OOO_LB_TC case
Bugzilla: 1462433 1499362
RH-Acked-by: Jerry Snitselaar <jsnitsel@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

commit 827d240a232d27cc12e9657d012f2e5ba953e98a
Author: Colin Ian King <colin.king@canonical.com>
Date:   Wed Apr 5 13:35:44 2017 +0100

    qed: fix missing break in OOO_LB_TC case

    There seems to be a missing break on the OOO_LB_TC case, pq_id
    is being assigned and then re-assigned on the fall through default
    case and that seems suspect.

    Detected by CoverityScan, CID#1424402 ("Missing break in switch")

    Fixes: b5a9ee7cf3be1 ("qed: Revise QM cofiguration")
    Signed-off-by: Colin Ian King <colin.king@canonical.com>
    Acked-by: Yuval Mintz <Yuval.Mintz@cavium.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1462433

Signed-off-by: Donald Dutile <ddutile@redhat.com>
Signed-off-by: Rafael Aquini <aquini@redhat.com>
---
 drivers/net/ethernet/qlogic/qed/qed_ll2.c | 1 +
 1 file changed, 1 insertion(+)

Index: src/drivers/net/ethernet/qlogic/qed/qed_ll2.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_ll2.c	2017-12-01 12:57:21.699058830 +0100
+++ src/drivers/net/ethernet/qlogic/qed/qed_ll2.c	2017-12-01 12:57:21.748058176 +0100
@@ -1131,6 +1131,7 @@
 		break;
 	case OOO_LB_TC:
 		pq_id = qed_get_cm_pq_idx(p_hwfn, PQ_FLAGS_OOO);
+		break;
 	default:
 		pq_id = qed_get_cm_pq_idx(p_hwfn, PQ_FLAGS_OFLD);
 		break;
